src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateHoldingEventHandler.java

```diff
@@ -118,13 +123,22 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
                 }
                 holdingAsJson.put("id", (i == 0) ? holdingsId : UUID.randomUUID().toString());
                 holdingAsJson.put("sourceId", FOLIO_SOURCE_ID);
-                fillInstanceIdIfNeeded(dataImportEventPayload, holdingAsJson);
+                fillInstanceIdIfNeeded(instanceId, holdingAsJson);
               }
 
               LOGGER.trace(format("handle:: Mapped holdings: %s", holdingsList.encode()));
               dataImportEventPayload.getContext().put(HOLDINGS.value(), holdingsList.encode());
               return List.of(Json.decodeValue(payloadContext.get(HOLDINGS.value()), HoldingsRecord[].class));
             })
+            .compose(holdingsToCreate -> consortiumService.getConsortiumConfiguration(context)
+              .compose(consortiumConfigurationOptional -> {
+                if (consortiumConfigurationOptional.isPresent()) {
+                  return ConsortiumUtil.createShadowInstanceIfNeeded(consortiumService, storage.getInstanceCollection(context),
+                    context, getInstanceId(dataImportEventPayload), consortiumConfigurationOptional.get()).map(holdingsToCreate)
+                    .map(holdingsToCreate);
```

Comment:
looks like `map(holdingsToCreate)` is called twice. Is this neccessary?

Reply:
Removed, thanks